hugr/src/hugr/serialize.rs

```diff
@@ -263,9 +256,15 @@ impl TryFrom<SerHugrV1> for Hugr {
             );
         }
 
-        for (node, metadata) in metadata.into_iter().enumerate() {
-            let node = portgraph::NodeIndex::new(node);
-            hugr.metadata[node] = metadata.as_object().cloned();
+        if let Some(metadata) = metadata {
+            for (node, metadata) in metadata
+                .into_iter()
+                .enumerate()
+                .filter_map(|(i, mb_metadata)| mb_metadata.map(|metadata| (i, metadata)))
+            {
+                let node = portgraph::NodeIndex::new(node);
+                hugr.metadata[node] = Some(metadata);
+            }
```

Comment:
This is a bit hard to read. Perhaps this is clearer?
```suggestion
            for (node, metadata) in metadata.into_iter().enumerate() {
                if let Some(metadata) = metadata {
                    let node = portgraph::NodeIndex::new(node);
                    hugr.metadata[node] = Some(metadata);
                }
            }
```